### PR TITLE
add CVE-2016-6795 curated data

### PR DESCRIPTION
**Pull request by camilo86**

### CODE REVIEW

File: cves/CVE-2016-6795.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: The URL provided by the user was not fully validated which could cause the server to let an attacker go to locations not permitted to view and even inject code to execute remotely
```
Comment (MattThyng) [inline cves/CVE-2016-6795.yml]:
```suggestion
description: |
   The URL provided by the user was not fully validated which could cause the server to let an attacker go to locations 
   not permitted to view and even inject code to execute remotely
```
How could the crafted URL remotely execute code? A little more detail here may be nice.

File: cves/CVE-2016-6795.yml [hunk 2]
```diff
@@ -209,4 +209,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: This looks like a coding mistake. Validating a valid secure URL is hard, but something as checking ../../.. should one of the first things the teams should have checked before upadating the validator. I can tell from one other commit that code is merged to master without being fully checked (commit can be found in interesting commits section)
```
Comment (MattThyng) [inline cves/CVE-2016-6795.yml:216]: What can be done to prevent future coding mistakes?

File: cves/CVE-2016-6795.yml (hunk 1 above)
Comment (PaulGildehaus) [reply cves/CVE-2016-6795.yml]: Mainly just agreeing with Matt about the change to the formatting. Adding the pipe character will make this much easier to read.

File: cves/CVE-2016-6795.yml [hunk 3]
```diff
@@ -69,9 +69,9 @@ fixes:
    - commit: c1869f4989942dd33fa4e189e0ac1f766fb5ac14
      note: Fix in trunk
 vccs:
-  - commit:
+  - commit: 8e67b9144aa643769b261e2492cb561e04d016ab
     note:
-  - commit:
+  - commit: 352fdb530080d04c6d1176c9d910120160000a6b
```
Comment (PaulGildehaus) [inline cves/CVE-2016-6795.yml:76]: What was the commit message / was something specifically broken by this commit, or was it just implemented wrong initially?

File: cves/CVE-2016-6795.yml [hunk 4]
```diff
@@ -148,8 +148,8 @@ interesting_commits:
     by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-    - commit:
-      note:
+    - commit: 83bb64f4c232348c4162ad2272c7eae1855a7362
+      note: Commit says new issues were introduced by last commit. Maybe workflow is not good at testing and verifying code?
```
Comment (PaulGildehaus) [inline cves/CVE-2016-6795.yml]: Were there commits after this commit fixing said issue?

File: cves/CVE-2016-6795.yml [hunk 5]
```diff
@@ -196,8 +196,8 @@ lessons:
     applies:
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: URL input can be formatted in many ways and validating each different section of it correctly can be hard. In this case the validator did not do the job right in every case allowing attacker to traverse through the whole server
```
Comment (PaulGildehaus) [inline cves/CVE-2016-6795.yml:204]: I would add the pipe character "|" so you can split this message into multiple with newline characters to make it more readable.

File: cves/CVE-2016-6795.yml [hunk 6]
```diff
@@ -178,8 +178,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: URL input was treated without thorough validatation which allow attackers to take advatange of this weakness and possibly access private files and/or run dangerous code
```
Comment (PaulGildehaus) [inline cves/CVE-2016-6795.yml:186]: I would add the pipe character "|" so you can split this message into multiple with newline characters to make it more readable.

**Comment by PaulGildehaus** [comment]
2 Upvotes for an interesting vulnerability.

**Pull request merged**